src/test/unit/controls/Graph/GraphAxes-spec.js

```diff
@@ -685,4 +693,205 @@ describe("Graph - Axes", () => {
             });
         });
     });
+    describe("Axis label", () => {
+        beforeEach(() => {
+            graph.destroy();
+            axisDefault.y2 = {
+                show: true,
+                label: "Some Y2 Label",
+                lowerLimit: 0,
+                upperLimit: 20
+            };
+        });
+        afterEach(() => {
+            axisDefault.y2 = "";
+        });
+        it("Truncates if too long", (done) => {
+            const labelAxisObj = utils.deepClone(axisDefault);
+            labelAxisObj.y.label = largeLabel;
+            labelAxisObj.y2.label = largeLabel;
+            labelAxisObj.x.label = largeLabel;
+            graph = new Graph(getAxes(labelAxisObj));
+            expect(
+                fetchElementByClass(styles.axisLabelX).querySelector("text")
+                    .textContent
+            ).toBe(
+                "Project long display value which is only for testing, Project long display value which is only for testing Project lon..."
+            );
+            expect(
+                fetchElementByClass(styles.axisLabelY).querySelector("text")
+                    .textContent
+            ).toBe("Project long display value w...");
+            expect(
+                fetchElementByClass(styles.axisLabelY2).querySelector("text")
+                    .textContent
+            ).toBe("Project long display value w...");
+            done();
+        });
+        it("Does not truncate if not long", (done) => {
+            graph = new Graph(getAxes(axisDefault));
+            expect(
+                fetchElementByClass(styles.axisLabelY).querySelector("text")
+                    .textContent
+            ).toBe("Some Y Label");
+            expect(
+                fetchElementByClass(styles.axisLabelX).querySelector("text")
+                    .textContent
+            ).toBe("Some X Label");
+            expect(
+                fetchElementByClass(styles.axisLabelY2).querySelector("text")
+                    .textContent
+            ).toBe("Some Y2 Label");
+            done();
+        });
+    });
```

Comment:
Looks like these tests repeated below also? Can we remove these?

Reply:
Actually we do need test that label do not truncate in that particular width and height. That is the reason why i repeated these in every case.